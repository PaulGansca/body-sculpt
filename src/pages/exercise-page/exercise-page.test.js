import { shallow } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import ExercisePage from './exercise-page';
import ExercisePageContainer from './exercise-page-container';

let wrapper;

describe('<ExercisePage />', () => {
    it("container renders exercise based on id", () => {
        const mockProps = {
            match: {params: {exerciseId: 1}},
            getState: jest.fn(),
            subscribe: jest.fn(),
            dispatch: jest.fn(),
            exercises: [{db_id: 1}],
        }
        wrapper = shallow(
                    <Provider store={mockProps}>
                        <ExercisePageContainer {...mockProps} />
                    </Provider>)
        expect(wrapper).toMatchSnapshot();
    })
    it("expect to render exercise page with props", () => {
        const mockProps = {
            exercise: {name: "Bench Press"},
        }
        wrapper = shallow(<ExercisePage {...mockProps} />)
        expect(wrapper).toMatchSnapshot();
    })
});