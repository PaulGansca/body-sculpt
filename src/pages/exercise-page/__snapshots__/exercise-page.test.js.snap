// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExercisePage /> container renders exercise based on id 1`] = `ShallowWrapper {}`;

exports[`<ExercisePage /> expect to render exercise page with props 1`] = `ShallowWrapper {}`;
